### ISSUE TITLE
Does not work on Samsung w/ OneUI 3.0 / Android 11

### ISSUE THREAD

**Issue by Depau** [magnusja/libaums#293]
Forwarding here from https://github.com/EtchDroid/EtchDroid/issues/117

### Problem

The affected users seem to be running Samsung's downstream ROM "OneUI" version 3.0 or later. I was able to borrow a devices with OneUI 1.0 and it doesn't seem to be affected.

I'm currently testing on a device with OneUI 2.1, though I'm having issues with wireless ADB and I will have to return it soon.

### Actual behavior

No user provided a logcat but the behavior seems the same as when the app is closed, then re-started with the same USB drive selected without unplugging and re-plugging the USB drive (ANR then timeout)

#### Stacktrace of Excpetion

```
(if exists)
```

**Comment by magnusja** [comment]
Interesting, I read the thread in EtchDroid/EtchDroid#117 but I am as confused as you probably are. 

I am not sure if we are missing something to make this fully compatible with Android 11?

I am assuming you are using the libusb communication? Maybe you want to package an app for the guys in EtchDroid/EtchDroid#117 using the vanilla Android USB API and see if that makes any difference?

**Comment by Depau** [comment]
> I am assuming you are using the libusb communication? Maybe you want to package an app for the guys in EtchDroid/EtchDroid#117 using the vanilla Android USB API and see if that makes any difference?

That's not a bad idea actually, I didn't think about it. No one provided logs though, it's hard to tell what's breaking exactly.

I just posted this here hoping you have or know someone who has a recent Samsung, it's hard to borrow devices around these times :)

**Comment by shirt-dev** [comment]
> > I am assuming you are using the libusb communication? Maybe you want to package an app for the guys in [EtchDroid/EtchDroid#117](https://github.com/EtchDroid/EtchDroid/issues/117) using the vanilla Android USB API and see if that makes any difference?
> 
> That's not a bad idea actually, I didn't think about it. No one provided logs though, it's hard to tell what's breaking exactly.
> 
> I just posted this here hoping you have or know someone who has a recent Samsung, it's hard to borrow devices around these times :)

It recently worked for me on my Samsung Galaxy S20 FE running OneUI 3.0 with no issues. If you need any testing let me know.

**Comment by Steve-George** [comment]
I am using MIUI in Redmi Note 10 Pro. This software cannot function well, too. It always reports an error and then screen will turn to    
black.

**Comment by Moohee-yll** [comment]
I was very troubled with the same problem. I found this is caused by function:  claimInterface (UsbInterface intf, **boolean force**). 

- If I let force == true (just like in libaums), then the crash appeared. 
- If I let force == false, this problem disappeared but the return value of  "claimInterface" is false, so I was unable to continue the next step.

So I this the key point is claimInterface( ).But I don't know how to do.

**Comment by steroidraging** [comment]
Are you trying this with or without OTG storage enabled in settings? 

Something I noticed, using my OnePlus Nord N10 5G is that for some reason with that enabled as you would typically need for the phone to notice the drive right, it actually throws off EtchDroid on my device. 

One day I tried EtchDroid and actually had forgotten to enable that option in System Settings and even tho most apps won't even recognize the drive without that enabled, it was the ONE thing that actually helped EtchDroid to fully complete the writing of the img successfully.

**Comment by depau** [comment]
@steroidraging I just noticed your reply. This is interesting, do you still have that device? I've never seen any phone with such an option.

**Comment by Ben-bit-code208** [comment]
i don't know if this issue is still active but i hope i have a fix for this i am not sure but i will open a PR in the hope that it will work i can't test it because i can't ADB into my phone (which is affected from this) and the issue was that the timeout for the retry was just ignored by some EMUI and OneUI 3.0+ device kernels PR will be shortly provided

**Comment by depau** [comment]
@Ben-bit-code208 if you open a PR feel free to add me as a reviewer, I can take a look

**Comment by Ben-bit-code208** [comment]
yeah i'm on it i just can't push to github because my .git folder is gone but i'm on it

**Comment by Ben-bit-code208** [comment]
okay i have the PR but @depau how can i add you as a reviewer?

**Comment by Ben-bit-code208** [comment]
#436

**Comment by depau** [comment]
Don't worry, I'll take a look when I have time

**Comment by Ben-bit-code208** [comment]
Okay

**Comment by mikenorthern96-stack** [comment]
I have built a complete OS with nothing more than my Android phone, and I am trying to use Etchdroid to make my iso image and it says it has GitHub issues

**Comment by magnusja** [comment]
> I have built a complete OS with nothing more than my Android phone, and I am trying to use Etchdroid to make my iso image and it says it has GitHub issues

I dont follow, can you elaborate.

**Comment by depau** [comment]
> > I have built a complete OS with nothing more than my Android phone, and I am trying to use Etchdroid to make my iso image and it says it has GitHub issues
> 
> I dont follow, can you elaborate.

They spammed this message in many etchdroid issues as well, I blocked them

**Comment by mikenorthern96-stack** [comment]
When etchdroid seen what it had written an iso for it got scared and said this is above our pay grade and GitHub banned me from Etch droid

**Comment by mikenorthern96-stack** [comment]
I was banned from etch droid by git hub after they wrote the ISO with the excuse of it's above their pay grade

**Comment by mikenorthern96-stack** [comment]
If indeed you have everything need to complete this which now that I've seen the problem I'm not sure because in the etchdroid had only written part of it and like Rufus it doesn't notify of problems and skips important partsand you end up mad
